tests/test_features/test_base.py

```diff
@@ -18,129 +18,444 @@
+##########################################################################
+
+@pytest.mark.usefixtures('discrete')
+class TestMultiFeatureVisualizer(object):
+    """
+    Test the MultiFeatureVisualizer base class
+    """
+
+    def test_subclass(self):
+        """
+        Check the visualizer/transformer class hierarchy
+        """
+        visualizer = MultiFeatureVisualizer()
+        assert isinstance(visualizer, FeatureVisualizer)
+        assert isinstance(visualizer, TransformerMixin)
+        assert isinstance(visualizer, BaseEstimator)
+        assert isinstance(visualizer, Visualizer)
+
+    def test_user_supplied_features(self):
+        """
+        Test that a user can supply feature names directly
+        """
+        X, y = self.discrete
+        features = ["f{}".format(i+1) for i in range(X.shape[1])]
+        oz = MultiFeatureVisualizer(features=features)
+
+        assert not hasattr(oz, 'features_')
+        assert oz.fit(X, y) is oz
+        assert hasattr(oz, 'features_')
+        npt.assert_array_equal(oz.features_, np.asarray(features))
+
+    def test_numeric_features(self):
+        """
+        Test that the features are column indices for numpy arrays
+        """
+        X, y = self.discrete
+        oz = MultiFeatureVisualizer()
+
+        assert not hasattr(oz, 'features_')
+        assert oz.fit(X, y) is oz
+        assert hasattr(oz, 'features_')
+        assert len(oz.features_) == X.shape[1]
+
+    @pytest.mark.skipif(pd is None, reason="test requires pandas")
+    def test_pandas_string_columns(self):
+        """
+        Ensure that DataFrame column names are uses as features
+        """
+        X, y = self.discrete
+        features = ["f{}".format(i+1) for i in range(X.shape[1])]
+
+        X = pd.DataFrame(X, columns=features)
+        y = pd.Series(y, name="target")
+
+        oz = MultiFeatureVisualizer()
+        assert not hasattr(oz, 'features_')
+        assert oz.fit(X, y) is oz
+        assert hasattr(oz, 'features_')
+        npt.assert_array_equal(oz.features_, np.asarray(features))
+
+    @pytest.mark.skipif(pd is None, reason="test requires pandas")
+    def test_pandas_no_columns(self):
+        """
+        Test when a DataFrame has no column names
+        """
+        """
+        Ensure that Pandas column names are uses as features
+        """
+        X, y = self.discrete
+
+        X = pd.DataFrame(X)
+        y = pd.Series(y, name="target")
+
+        oz = MultiFeatureVisualizer()
+        assert not hasattr(oz, 'features_')
+        assert oz.fit(X, y) is oz
+        assert hasattr(oz, 'features_')
+        assert len(oz.features_) == X.shape[1]
+
 
 ##########################################################################
 ## DataVisualizer Tests
 ##########################################################################
 
 @pytest.mark.usefixtures("discrete", "continuous")
-class TestDataVisualizerBase(VisualTestCase):
+class TestDataVisualizer(object):
```

Comment:
DataVisualizer now does so much work that I believed it was critical for it to be fully tested. I would welcome an analysis of the coverage of the tests here since a number of downstream visualizers will rely on this class including:

- PCA
- Manifold
- ParallelCoordinates
- RadViz

In particular, I'm worried about my coverage of user parameters (e.g. color/colormap, etc.) - though the defaults are well tested.